search/includes/classes/class-settingshealthjob.php

```diff
@@ -200,4 +220,84 @@ public function send_alert( $channel, $message, $level, $type = '' ) {
 
 		return wpcom_vip_irc( $channel, $message, $level );
 	}
+
+	/**
+	 * Determine whether to build new index as part of auto healing.
+	 *
+	 * @param object $indexable The Indexable we want to rebuild.
+	 */
+	public function maybe_build_new_index( $indexable ) {
+		// Only do for non-production for now.
+		$is_prod = defined( 'VIP_GO_APP_ENVIRONMENT' ) && 'production' === VIP_GO_APP_ENVIRONMENT;
+		if ( $is_prod ) {
+			return;
+		}
+
+		// Check if lock is set. Bail if new build is already ongoing.
+		$new_index_lock = get_option( 'vip_search_new_version_building_for_shards' );
+		if ( false === $new_index_lock ) {
+			// No lock, check if event has been scheduled already.
+			if ( ! wp_next_scheduled( self::CRON_EVENT_BUILD_NAME ) ) {
+				$scheduled_event = wp_schedule_single_event( time() + 30, self::CRON_EVENT_BUILD_NAME, [ $indexable ] );
+
+				if ( is_wp_error( $scheduled_event ) ) {
+					// Send alert that failed to schedule event.
+					$error_message = $scheduled_event->get_error_message() ?? 'Event could not be scheduled';
+					$message       = sprintf( 'Failed to schedule event for building new %s index version on %s to meet shard requirements: %s', $indexable->slug, home_url(), $error_message );
+					$this->send_alert( '#vip-go-es-alerts', $message, 2 );
+				}
```

Comment:
Gotta pass the 4th flag to `wp_schedule_single_event` to get a WP Error back on failure: https://developer.wordpress.org/reference/functions/wp_schedule_single_event/

Honestly though this is very very unlikely, I'd personally just drop the extra error handling / alerting here.

Reply:
Good catch and suggestion noted!